### ISSUE TITLE
On new installations you run into nginx permission errors by default

### ISSUE THREAD

**Issue by KDederichs** [mastodon/mastodon#27223]
### Steps to reproduce the problem

After having to set up my server again, I did the steps in https://docs.joinmastodon.org/admin/install/ again and ran into an issue with nginx permissions again (like the first time, which I forgot about till now).

It seems like by default, following the installation steps you'll run into nginx errors like this one:
```
2023/09/30 13:38:38 [crit] 69673#69673: *1375 stat() "/home/mastodon/live/public/500.html" failed (13: Permission denied), client: 172.71.130.175, server: XXXXXXX, request: "GET /sw.js HTTP/2.0"
```
which indicates that the web server can not access the static assets.

It's easily fixable by adding the nginx user (www-data) to the mastodon group but sadly that's not mentioned in the docs, so I thought I'd mention this

### Expected behaviour

No errors

### Actual behaviour

Access denied errors

### Mastodon instance

/

### Mastodon version

v4.2.0

### Browser name and version

Firefox 118

### Operating system

OSX14

**Comment by 33b5e5** [comment]
This is still an issue in 2024. Was banging my head against a wall until I found this. This command resolves the issue:

`sudo usermod -aG mastodon www-data`

**Comment by lwndow** [comment]
> This is still an issue in 2024. Was banging my head against a wall until I found this. This command resolves the issue:
> 
> `sudo usermod -aG mastodon www-data`

Same for me on a fresh install today, but makes sense, since user directories (such as mastodon) are created with default perms of 700 (for user privacy). The group perms suggested here would work if the group had read-access (you'd need to `chown mastdon:www-data /home/mastodon` and `chmod` to at least 750 or possibly 751 for systemd access in my testing). 

Something that seems simpler is to `chmod 701 /home/mastdon` to just enable the execute bit for world. This allows my instance to stop producing errors about permissions. I'm not sure this is safer than adding group-read, but it feels like it should be?

**Comment by 33b5e5** [comment]
> The group perms suggested here would work if the group had read-access (you'd need to `chown mastdon:www-data /home/mastodon` and `chmod` to at least 750 or possibly 751

No, I don't think that is correct. The command provided -- `sudo usermod -aG mastodon www-data` -- adds the user www-data (which nginx uses) to the group mastodon. I think you read it backwards?

Since the /home/mastodon directories should already have read-access via the mastodon group, no chmod is needed after.

**Comment by lwndow** [comment]
> I think you read it backwards?

It is possible I read it wrong and misunderstood, but the group change and restart of services didn't seem to work for me today. I'm likely missing something, but even with group access when the parent directory has no group permissions, I don't see how it can access? Are your `/home/mastodon` perms also 700? Would you mind sharing to see where i might have gone astray from yours?

**Comment by lwndow** [comment]
For posterity I started over in my VPS (Debian Bookworm), documenting here for others.

After the user creation, default directory perms:
```
# ls -ld /home/mastodon
drwx------ 9 mastodon mastodon 4096 Jun  1 21:43 /home/mastodon
```

After installation, but before starting services, I updated group membership:
```
$ groups www-data
www-data : www-data mastodon
$ ps auxwww | grep nginx
root         542  0.0  0.3  32292  6888 ?        Ss   20:42   0:00 nginx: master process /usr/sbin/nginx -g daemon on; master_process on;
www-data   15905  0.0  0.4  33688  8468 ?        S    21:45   0:00 nginx: worker process
www-data   15906  0.0  0.2  33552  4404 ?        S    21:45   0:00 nginx: cache manager process
```

This _does_ allow mastodon-web service to start, but I still see permission errors in the error log:
```
2024/06/01 21:45:24 [crit] 15905#15905: *59 stat() "/home/mastodon/live/public/packs/js/application-de50c0d962d466c06dca.chunk.js" failed (13: Permission denied), client: $ipv6-address, server: $server, request: "GET /packs/js/application-de50c0d962d466c06dca.chunk.js HTTP/2.0", host: "$server", referrer: "https://$server/"
```

Updating the directory to add group permissions fixes the issue for me: `chmod 750 /home/mastodon`, or, alternatively the way I had it originally. Anyone's mileage may vary.

**Comment by mjankowski** [comment]
I believe this was closed by https://github.com/mastodon/documentation/pull/1537 by not auto-resolved.

Closing now, but please reopen if I'm missing something!
